pages/index.tsx

```diff
@@ -0,0 +1,174 @@
+import { Inter } from "next/font/google";
+import { useQuery } from "react-query";
+import { getWords } from "../services/getWord";
+import { useState } from "react";
+import Keyboard from "../components/Keyboard/Keyboard";
+import { endOfArray } from "@/utils/findEndOfArray";
+import ResultsLetters from "../components/ResultsLetters/ResultsLetters";
+import toast from "react-hot-toast";
+import Modal from "@/components/Modal";
+
+const inter = Inter({ subsets: ["latin"] });
+
+export enum Status {
+  FOUND = "FOUND",
+  IN_THE_WRONG_PLACE = "IN_THE_WRONG_PLACE",
+  MISS = "MISS",
+  IN_TRY = "IN_TRY",
+}
+
+enum GameEndStatus {
+  WIN = "WIN",
+  LOST = "LOST",
+}
+
+type IsEnd = { isEnd: boolean; status: GameEndStatus };
+export type Word = { letter: string; status: Status }[];
+
+const Home = () => {
+  const [words, setWords] = useState<Word[]>([[]]);
+  const [isEnd, setIsEnd] = useState<IsEnd>();
+
+  const {
+    data: answerWord,
+    isError,
+    isLoading,
+  } = useQuery({
+    queryFn: () => getWords(),
+    queryKey: ["words"],
+  });
+
+  const { data: checkWord } = useQuery({
+    queryFn: () =>
+      getWords(
+        words[endOfArray(words).outer]
+          .map((e) => e.letter)
+          .join("")
+          .toLowerCase()
+      ),
+    queryKey: [
+      "words",
+      words[endOfArray(words).outer]
+        .map((e) => e.letter)
+        .join("")
+        .toLowerCase(),
+    ],
+  });
```

Comment:
Even if we assume that this endpoint has all the correct Polish words, if the user has a bad internet connection or no internet connection at all, they will receive an error because the request will not complete in time. This request should be awaited in checkLastWordInArray function.